### ISSUE TITLE
Protocol class type representation

### ISSUE THREAD

**Issue by kaos** [python/mypy#10988]
**Bug Report**

**To Reproduce**

`repro.py`:
```python
from dataclasses import dataclass
from typing import Generic, Protocol, Type, TypeVar

T = TypeVar("T")

class MyProto(Protocol):
    attr: str

@dataclass
class Impl:
    attr: str

@dataclass
class Demo(Generic[T]):
   proto: T
   impl: Type[T]

wrong_T = Demo(MyProto, Impl)
wrong_T = 1  # to expose inferred type of T (which is `object`)

wrong_arg1 = Demo[MyProto](MyProto, Impl)
```

**Expected Behavior**

I expected there be a type representing a Protocol class.

I expected `T` to be inferred as `MyProto` (as `Type[MyProto]` is for types implementing the protocol, not the protocol type itself.) for the `wrong_T` variable. Since `Type[T]` should accept any object when `T` is a protocol, and then if that object does not satisfy the Protocol, that is an error.

Moreover, for `wrong_arg1`, I expected `MyProto` to be an acceptable argument for type `MyProto`, but its type is reported as `Type[MyProto]`, but that does not fly, as if that type is expected, and I pass `MyProto`, it complains (rightfully so):
> t.py:24:34: error: Only concrete class can be given where "Type[MyProto]" is expected  [misc]

**Actual Behavior**

```bash
$ python -m mypy repro.py
t.py:22:11: error: Incompatible types in assignment (expression has type "int",
variable has type "Demo[object]")  [assignment]
    wrong_T = 1  # to expose inferred type of T
              ^
t.py:24:28: error: Argument 1 to "Demo" has incompatible type "Type[MyProto]";
expected "MyProto"  [arg-type]
    wrong_arg1 = Demo[MyProto](MyProto, Impl)
                               ^
Found 2 errors in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: `0.910`
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: `Python 3.8.5`
- Operating system and version: MacOS 11.5.2, `Darwin Kernel Version 20.6.0`

Related StackOverflow question: https://stackoverflow.com/questions/68822297/protocol-implementation-class-type-check
